File: heitech.configXt.Models/UiOperationResult.cs
using System.Collections.Generic;

namespace heitech.configXt.Models
{
    ///<summary>
    /// Use Result for the Actual OperationResult
    ///</summary>
    public class UiOperationResult
    {
        ///<summary>
        /// For Serialization
        ///</summary>
        public UiOperationResult()
        {
        }

        ///<summary>
        /// For use inside the app
        ///</summary>
        public UiOperationResult(bool success, string resultType, string errorMessage, IEnumerable<ConfigurationModel> model)
        {
            Initialize(success, resultType, errorMessage, model);
        }

        private void Initialize(bool success, string resultType, string errorMessage, IEnumerable<ConfigurationModel> entity)
        {
            IsSuccess = success;
            ResultName = resultType;
            ConfigurationModels = new List<ConfigurationModel>();

            if (IsSuccess == false)
            {
                ErrorMessage = errorMessage;
                return;
            }
            ConfigurationModels = new List<ConfigurationModel>(entity);
        }

        ///<summary>
        /// All the Returned Models. Zero or one Item, except when ReadAll was used
        ///</summary>
        public List<ConfigurationModel> ConfigurationModels { get; set; }
        ///<summary>
        /// Indicates if the operation succedeed
        ///</summary>
        public bool IsSuccess { get; set; }
        ///<summary>
        /// If no Success, this is the error that occured
        ///</summary>
        public string ErrorMessage { get; set; }
        ///<summary>
        /// The ResultType of the Operation (think http status code)
        ///</summary>
        public string ResultName { get; set; }
    }

}

File: heitech.configXt.Client.Nodejs/src/uiOperationResult.ts
import { ConfigurationModel } from "./configModel";

export class UiOperationResult
{
    constructor(public IsSuccess:boolean,
                public ResultName:string, 
                public ErrorMessage:string, 
                public ConfigurationModels:ConfigurationModel[])
    { }

    public format() : string {
        let models : string = this.ConfigurationModels.map(c => `${c.Name} --> ${c.Value}`).join("|")
        return `Success:[${this.IsSuccess}]\nResult:[${this.ResultName}]\nerror:[${this.ErrorMessage}]\nModels:[${models}]`;
    }

    static FromMsg(result : any) : UiOperationResult{
        return new UiOperationResult(
            result.IsSuccess as boolean,
            result.ResultName as string,
            result.ErrorMessage as string,
            result.ConfigurationModels as ConfigurationModel[],
        );
    }
}